src/gl2d/serialize/serialize.ts

```diff
@@ -0,0 +1,48 @@
+import { type Renderer } from '../../rendering/renderers/types';
+import { type Container } from '../../scene/container/Container';
+import { type Gl2DFile, type ToGl2D } from '../file';
+import { deepRemoveUndefined } from '../utils/deepRemoveUndefined';
+
+/**
+ * Options for serializing a PixiJS v8 scene graph into a GL2D scene file.
+ * @category gl2d
+ * @standard
+ */
+export interface ToGl2DOptions
+{
+    gl2D?: ToGl2D;
+    renderer: Renderer;
+}
+
+/**
+ * Serialize a PixiJS v8 scene graph into a GL2D scene file.
+ * @param options - The options for serialization
+ * @param options.root - The root container for the active scene
+ * @param options.renderer - The renderer instance
+ * @returns The serialized GL2D JSON object
+ * @category gl2d
+ * @standard
+ */
+export async function serializeGl2D(options: { root: Container, renderer: Renderer }): Promise<Gl2DFile>
```

Comment:
Nit organizationally how do you feel about a GL2D object that has a serialize method instead of serializeGl2D? Like Assets. Would allow us to add more functionality in the future and reads a little easier.

Reply:
This is a good idea! I'll update the PR